### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by 11000100111000** [typeorm/typeorm#968]
```js
@Entity()
export class Token {
    @OneToOne(type => User, {primary: true})
    @JoinColumn({name: "email", referencedColumnName: "email"})
    public email: string;
    @Column()
    public token: string;
}

@Entity()
export class User {
    @PrimaryGeneratedColumn()
    public id: number;
    @Column({unique: true})
    public email: string;
}
```
Using these Entity-Definitions yields the following error during the second(and following) run (while trying to sync the schema). The first run works fine however.
```
ERROR:  column "email" is in a primary key
STATEMENT:  ALTER TABLE "public"."token" ALTER COLUMN "email" DROP NOT NULL
```
For whatever reason it tries to remove the Not Null from the Primary key. The Database we are using is Postgres

**Comment by 11000100111000** [comment]
My current workaround would be adding a nullable:false clause to the OneToOne decorator

**Comment by pleerock** [comment]
Fixed and will be released in `0.1.8`.
